src/Catel.MVVM/Catel.MVVM.Shared/Services/NavigationService.xamarin.forms.cs

```diff
@@ -54,39 +95,27 @@ public override bool CanGoForward
         /// <returns>The target to navigate to.</returns>
         protected override string ResolveNavigationTarget(Type viewModelType)
         {
-            var dependencyResolver = this.GetDependencyResolver();
-            var viewLocator = dependencyResolver.Resolve<IViewLocator>();
-
-            var navigationTarget = viewLocator.ResolveView(viewModelType).AssemblyQualifiedName;
-            return navigationTarget;
+            return _viewLocator.ResolveView(viewModelType).AssemblyQualifiedName;
         }
 
         /// <summary>
-        /// Gets the back stack count.
+        /// Returns the number of total back entries (which is the navigation history).
         /// </summary>
-        /// <returns>System.Int32.</returns>
         public override int GetBackStackCount()
         {
-            var backStackCount = 0;
-            var currentPage = Application.Current.CurrentPage();
-            if (currentPage != null)
-            {
-                backStackCount = currentPage.Navigation.ModalStack.Count - 1;
-            }
-
-            return backStackCount;
+            return Application.Current.GetActivePage().Navigation.NavigationStack.Count;
```

Comment:
Hell to debug this in case of a NullReferenceException. Please clean up / split up this code.